api/cloudcontroller/ccv3/paginate.go

```diff
@@ -69,3 +74,24 @@ func (requester RealRequester) wrapFirstPage(request *cloudcontroller.Request, o
 
 	return wrapper, warnings, nil
 }
+
+func (requester RealRequester) bulkRetrieval(request *cloudcontroller.Request, obj interface{}, appendToExternalList func(interface{}) error) (IncludedResources, Warnings, error) {
+	wrapper, warnings, err := requester.wrapFirstPage(request, obj, func(_ interface{}) error { return nil })
+	if err != nil {
+		return IncludedResources{}, warnings, err
+	}
+
+	newQuery := url.Values{}
+	for name, value := range request.URL.Query() {
+		if name != "" && name != string(PerPage) {
+			newQuery.Add(name, strings.Join(value, ","))
+		}
+	}
+	resultsPerPage := strconv.Itoa(wrapper.Pagination.TotalResults)
```

Comment:
**suggestion** I don't think theres any benefit to requesting the exact total results per page. We might as well just request the max (5000 per page) and if theres less than that thats fine.